src/app/components/modules/progress/progress.component.ts

```diff
@@ -240,19 +244,26 @@ export class ProgressComponent extends SubscriptionsComponent implements OnInit,
 			const currBlock = this.schedule.classes[i];
 			const nextBlock = this.schedule.classes[i + 1];
 
-			if (currBlock.end !== nextBlock.start) {
-				breaks.push({
-					class: {
-						name: 'Break',
-						teacher: null,
-						type: ClassType.OTHER,
-						block: Block.OTHER,
-						color: 'rgba(0, 0, 0, 0.4)',
-						textDark: false
-					},
-					start: currBlock.end,
-					end: nextBlock.start
-				});
+			console.log(this.schedule.classes[this.schedule.classes.length - 1].end !== SchoolDayEnd315);
+
+			var breakObj = {
+			 class: {
+				 name: 'Break',
+				 teacher: null,
+				 type: ClassType.OTHER,
+				 block: Block.OTHER,
+				 color: 'rgba(0, 0, 0, 0.4)',
+				 textDark: false
+			 },
+			 start: currBlock.end,
+			 end: nextBlock.start
+		 };
+
+			if (this.schedule.classes[this.schedule.classes.length - 1].end !== SchoolDayEnd315 && this.schedule.classes.length - 1 === i){// If there's a break as the last class of the day, make the break end at 3:15
```

Comment:
Typically we put comments on their own line. Maybe put it above the `if` statement?